addons/stock/models/stock_rule.py

```diff
@@ -472,7 +472,7 @@ def raise_exception(procurement_errors):
         return True
 
     @api.model
-    def _search_rule(self, route_ids, packaging_id, product_id, warehouse_id, domain):
+    def _search_rule(self, route_ids, packaging_id, product_id, warehouse_id, domain, carrier_id=None):
```

Comment:
It's not a good idea to add a kwargs in a module that has no link to it. Instead, I would either add a hook to call extra search and an override in delivery_stock. But it will be very limited since we can't move the priority. Otherwise you could add the shipping routes in the values during _prepare_procurement_values (easier)